src/psbt.h

```diff
@@ -462,8 +464,9 @@ struct PartiallySignedTransaction
                 break;
             }
 
-            // First byte of key is the type
-            unsigned char type = key[0];
+            // Type is compact size uint at beginning of key
+            VectorReader skey(s.GetType(), s.GetVersion(), key, 0);
+            uint64_t type = ReadCompactSize(skey);
```

Comment:
Manually inspected that all the fields with constant types are now being written with compact size, but I would like to be more systematic somehow in the future.